hibernate-core/src/main/java/org/hibernate/query/sqm/sql/internal/StandardSqmSelectTranslator.java

```diff
@@ -187,6 +195,24 @@ protected void prepareQuerySpec(QuerySpec sqlQuerySpec) {
 
 	@Override
 	protected void postProcessQuerySpec(QuerySpec sqlQuerySpec) {
+		final List<TableGroup> roots = sqlQuerySpec.getFromClause().getRoots();
+		if ( roots != null && roots.size() == 1 ) {
+			final TableGroup root = roots.get( 0 );
+			final ModelPartContainer modelPartContainer = root.getModelPart();
+			final EntityPersister entityPersister = modelPartContainer.findContainingEntityMapping().getEntityPersister();
+			assert entityPersister instanceof AbstractEntityPersister;
+			final String primaryTableAlias = root.getPrimaryTableReference().getIdentificationVariable();
```

Comment:
An enhancement here would be to support filtering on non-primary-table columns

Reply:
I will implement this in the PR for https://trello.com/c/lF9R13J3/115-where-and-wherejointable, which will be created pretty soon.

Reply:
Initially I thought this was easy to implement but then I found my code needs to invoke the following method from `org.hibernate.persister.entity.Joinable`.
```
	/**
	 * Get the where clause filter, given a query alias and considering enabled session filters
	 */
	public String filterFragment(String alias, Map<String, Filter> enabledFilters) throws MappingException;
```
seems only primary table alias is needed here. For that reason, I would be cautious to maintain the status quo and keep the code changes out of the new PR which has been created: https://github.com/hibernate/hibernate-orm/pull/3366 as I originally promised.

Plan to investigate further and will create separate PR if needed.